src/components/legend/legend.js

```diff
@@ -29,7 +34,11 @@ const LegendContainer = props => {
   const handleRemoveLayer = (layer) => {
     const { activeLayers, removeLayerAnalyticsEvent, changeGlobe } = props;
     if (layer.legendConfig.groupedLayer) {
-      batchLayerManagerToggle(LEGEND_GROUPED_LAYERS_GROUPS[layer.title], activeLayers, changeGlobe);
+      const activeGroupedLayers = intersection(
+        activeLayers.map((l) => l.title),
+        LEGEND_GROUPED_LAYERS_GROUPS[layer.title]
+      );
+      batchToggleLayers(activeGroupedLayers, activeLayers, changeGlobe);
```

Comment:
Maybe we can move this `intersection` logic into its own utils function?

in `utils/layer-manager-utils.js`

```javascript
      const getActiveLayersFromLayerGroup = (layerGroup, activeLayers) => (
        intersection(activeLayers.map((l) => l.title), layerGroup)
      );
```
in `components/legend`

```javascript  
      const activeGroupedLayers = getActiveLayersFromLayerGroup(LEGEND_GROUPED_LAYERS_GROUPS[layer.title], activeLayers);
      batchToggleLayers(activeGroupedLayers, activeLayers, changeGlobe);
```